src/core/net/ip6.cpp

```diff
@@ -1035,8 +1036,8 @@ Error Ip6::ProcessReceiveCallback(Message &          aMessage,
             Udp::Header udp;
 
             IgnoreError(aMessage.Read(aMessage.GetOffset(), udp));
-            VerifyOrExit(Get<Udp>().ShouldUsePlatformUdp(udp.GetDestinationPort()), error = kErrorNoRoute);
-
+            VerifyOrExit(Get<Udp>().ShouldUsePlatformUdp(udp.GetDestinationPort())
+                         && !Get<Udp>().IsPortInUse(udp.GetDestinationPort()), error = kErrorNoRoute);
```

Comment:
Here the new code blocks the message to be passed to the TUN interface (i.e. wpan0), so we need to pass it to the socket `Udp::HandlePayload` in `Udp::HandleMessage`.

Unfortunately, when `PLATFORM_UDP` is used, the following code will block the message:

https://github.com/openthread/openthread/blob/713231e144c9ccd46126323fb14d2ab06cdb6361/src/core/net/udp6.cpp#L479-L481

I think we need to allow UDP messages for OpenThread in-use ports be passed to `Udp::HandlePayload`.

@bukepo Do you have any other thoughts on this issue?